flytekit/control_plane/nodes.py

```diff
@@ -195,9 +200,13 @@ def inputs(self) -> Dict[str, Any]:
                         _common_utils.load_proto_from_file(_literals_pb2.LiteralMap, tmp_name)
                     )
 
-            # TODO: need to convert flyte literals to python types. For now just use literals
-            # self._inputs = TypeEngine.literal_map_to_kwargs(ctx=FlyteContext.current_context(), lm=input_map)
-            self._inputs = input_map
+            task_id = self.task_executions[0].id.task_id
+            task = FlyteTask.fetch(task_id.project, task_id.domain, task_id.name, task_id.version)
+            self._inputs = TypeEngine.literal_map_to_kwargs(
```

Comment:
its not required to guess the types always right? We can provide the actual definition and then guessing is not required

Reply:
what's the actual definition?